modules/node/src/listener/listener.service.ts

```diff
@@ -134,26 +134,53 @@ export default class ListenerService implements OnModuleInit {
         this.logEvent(UNINSTALL_EVENT, data);
         // check if app being uninstalled is a receiver app for a transfer
         // if so, try to uninstall the sender app
-        this.transferService.reclaimLinkedTransferCollateralByAppInstanceId(
-          data.data.appInstanceId,
-        );
+        // TODO: i think there are race conditions here that are causing errors
+        // let's reenable this when we figure it out better. for now, the node will
+        // reclaim on client check-in
+        // await this.transferService.reclaimLinkedTransferCollateralByAppInstanceIdIfExists(
+        //   data.data.appInstanceId,
+        // );
       },
       UNINSTALL_VIRTUAL_EVENT: (data: UninstallVirtualMessage): void => {
         this.logEvent(UNINSTALL_VIRTUAL_EVENT, data);
       },
       UPDATE_STATE_EVENT: async (data: UpdateStateMessage): Promise<void> => {
         // if this is for a recipient of a transfer
         this.logEvent(UPDATE_STATE_EVENT, data);
-        const { newState } = data.data;
+        const { newState, appInstanceId } = data.data;
         let transfer = await this.linkedTransferRepository.findByLinkedHash(
           (newState as SimpleLinkedTransferAppState).linkedHash,
         );
         if (!transfer) {
-          this.log.debug(`Could not find transfer for update state event`);
+          this.log.debug(
+            `Could not find transfer for update state event for app: ${appInstanceId}`,
+          );
+          return;
+        }
+        if (appInstanceId !== transfer.receiverAppInstanceId) {
+          this.log.debug(
+            `Not updating transfer preimage or marking as redeemed for sender update state events`,
+          );
           return;
         }
         // update transfer
         transfer.preImage = (newState as SimpleLinkedTransferAppState).preImage;
+
+        if (
+          transfer.status === LinkedTransferStatus.RECLAIMED ||
+          transfer.status === LinkedTransferStatus.REDEEMED
+        ) {
+          this.log.warn(
+            `Got update state event for a receiver's transfer app (transfer.id: ${transfer.id}) with unexpected status: ${transfer.status}`,
+          );
+          return;
+        }
+
+        // transfers are set to `PENDING` when created. They are set to
+        // `FAILED` when the receiver rejects an install event. If a transfer
+        // makes it to the `UPDATE_STATE_EVENT` portion, it means the transfer
+        // was successfully installed. There is no reason to not redeem it in
+        // that case.
```

Comment:
we could probably also just make the condition
```typescript
if (transfer.status !== LinkedTransferStatus.PENDING) { ... }
```

but we don't technically *have* to